src/lib/utils.c

```diff
@@ -483,7 +483,7 @@ int match_string(const char *string, const char *pattern)
 
 	status = regcomp(&regex, pattern, REG_EXTENDED);
 	if (status != 0) {
-		printf("regecomp failed, ret=%d", status);
+		lib_log(ctx, LED_LOG_LEVEL_ERROR, "regecomp failed, ret=%d", status);
```

Comment:
This message doesn't seem to say enough about the problem, maybe it's worth changing it to something like this right away.
("Failed to initialize regex with pattern %s. regcomp status %d.", pattern, status)

Reply:
Done as requested, thanks for the review.